### ISSUE TITLE
Improve performance of js-config-webpack-plugin

### ISSUE THREAD

**Issue by ernscht** [namics/webpack-config-plugins#24]
## I'm submitting a...

<pre><code>
[x] Performance improvement
</code></pre>

## Current & Expected behavior

Without the [thread-loader](https://github.com/namics/webpack-config-plugins/blob/master/packages/js-config-webpack-plugin/config/development.config.js#L16r), compilation takes three to four times less time on changes. We could remove it.

## What is the motivation / use case for changing the behavior?

At least in the [nitro example](https://github.com/namics/generator-nitro/tree/master/packages/project-nitro) it works much faster without the thread-loader.

The "thread-loader" could be useful for a lot of files or with a different configuration (prewarming). Maybe even a configuration option makes sense?

**Comment by jantimon** [comment]
I wrote a small node file to measure the behaviour.

The most important part are incremental build times.  
The tool does the following:

![measure-flow](https://user-images.githubusercontent.com/4113649/50090658-8f152680-0209-11e9-86a3-036857f58d85.png)

Result:

```
current = ts-loader
latest = cache-loader + thread-loader + ts-loader

⏰ DevServer Current { 
  lowest: 0.383,
  highest: 0.728,
  average: 0.47200000000000003,
  median: 0.478 
} 
⏰ DevServer Latest { 
  lowest: 1.1280000000000001,
  highest: 1.46,
  average: 1.2519310344827583,
  median: 1.232 
}
```

The second test is running multiple full builds without dev server on the same machine:

```
current = ts-loader
latest = cache-loader + thread-loader + ts-loader

⏰ Build Current - with file cache { 
  lowest: 4.66,
  highest: 6.116,
  average: 5.0339,
  median: 4.8629999999999995 
} 
⏰ Build Latest - with file cache { 
  lowest: 3.545,
  highest: 4.473,
  average: 3.8223000000000007,
  median: 3.8135 
} 
```

The third test is running multiple full builds on a clean system (similar to a CI build):

```
current = ts-loader
latest = cache-loader + thread-loader + ts-loader

⏰ Build Current - clean build { 
  lowest: 11.971,
  highest: 13.212,
  average: 12.3424,
  median: 12.3425 
} 
⏰ Build Latest - clean build { 
  lowest: 12.226,
  highest: 16.078,
  average: 14.0427,
  median: 14.6725
} 
```

**Comment by jantimon** [comment]
Same result on travis:

Removing thread-loader improved my build speed by 65% for ts-loader and 65% for babel-loader:
https://travis-ci.org/namics/webpack-config-plugins/jobs/469599663
https://travis-ci.org/namics/webpack-config-plugins/jobs/469599664

**Comment by jantimon** [comment]
I could fix the thread-loader :tada:

The results look promising (comparing version thread-loader 1.2.0 and thread-loader 2.1.0):


#### Time difference for 100 components:

https://travis-ci.org/namics/webpack-config-plugins/jobs/470988476 (typescript)
https://travis-ci.org/namics/webpack-config-plugins/jobs/470988477 (babel)

 **1.171 s -> 378ms**

 average: '0.79s faster than the latest released npm version',
 averageRelative: '32% of the latest released npm version build duration',
 median: '0.79s faster than the latest released npm version',
 medianRelative: '32% of the latest released npm version build duration'

#### Time difference for 1000 components:

https://travis-ci.org/namics/webpack-config-plugins/jobs/470988478 (typescript)
https://travis-ci.org/namics/webpack-config-plugins/jobs/470988479 (babel)

 **3.071s -> 1.34s**

average: '1.72s faster than the latest released npm version',
 averageRelative: '44% of the latest released npm version build duration',
 median: '1.74s faster than the latest released npm version',
 medianRelative: '42% of the latest released npm version build duration'

**Comment by jantimon** [comment]
Fixed by https://github.com/namics/webpack-config-plugins/pull/33

**Comment by jantimon** [comment]
Released in 1.2.0
